snuba/datasets/outcomes.py

```diff
@@ -149,24 +170,38 @@ def __init__(self) -> None:
             dist_destination_table_name=READ_DIST_TABLE_NAME,
         )
 
-        dataset_schemas = DatasetSchemas(
-            read_schema=read_schema,
-            write_schema=write_schema,
-            intermediary_schemas=[materialized_view],
+        # The raw table we write onto, and that potentially we could
+        # query.
+        writable_storage = TableStorage(
+            schemas=StorageSchemas(read_schema=raw_schema, write_schema=raw_schema),
+            table_writer=TableWriter(
+                write_schema=raw_schema,
+                stream_loader=KafkaStreamLoader(
+                    processor=OutcomesProcessor(), default_topic="outcomes",
+                ),
+            ),
+            query_processors=[],
         )
-
-        table_writer = TableWriter(
-            write_schema=write_schema,
-            stream_loader=KafkaStreamLoader(
-                processor=OutcomesProcessor(), default_topic="outcomes",
+        # The materialized view we query aggregate data from.
+        materialized_storage = TableStorage(
+            schemas=StorageSchemas(
+                read_schema=read_schema,
+                write_schema=None,
+                intermediary_schemas=[materialized_view_schema],
             ),
+            query_processors=[PrewhereProcessor()],
+        )
+        storage_selector = OutcomesQueryStorageSelector(
+            raw_table=writable_storage, materialized_view=materialized_storage
         )
 
         super().__init__(
-            dataset_schemas=dataset_schemas,
-            table_writer=table_writer,
+            storages=[writable_storage, materialized_storage],
+            storage_selector=storage_selector,
+            abstract_column_set=read_schema.get_columns(),
+            writable_storage=writable_storage,
             time_group_columns={"time": "timestamp"},
-            time_parse_columns=("timestamp",),
+            time_parse_columns=("timestamp"),
```

Comment:
This doesn't look right. It probably type checks though, which is scary.

We should use lists instead of tuples here to avoid this kind of problem. If we need to imply immutability, we should be doing that in the type system instead of abusing tuples.

Reply:
yes, this is broken. why it passes the test specifically defined to test this is even more scary.